test/contlinear.jl

```diff
@@ -116,6 +116,178 @@ function contlineartest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64)
+        @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.GreaterThan}()) == 3
+
+        cf2 = MOI.ScalarAffineFunction(v, [1.0,1.0,1.0], 0.0)
+        MOI.modifyconstraint!(m, c, cf2)
+
+        objf = MOI.ScalarAffineFunction(v, [1.0,0.0,2.0], 0.0)
+        MOI.setobjective!(m, MOI.MaxSense, objf)
+
+        @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.LessThan}()) == 1
+        @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.GreaterThan}()) == 3
+
+        MOI.optimize!(m)
+
+        @test MOI.cangetattribute(m, MOI.TerminationStatus())
+        @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+
+        @test MOI.cangetattribute(m, MOI.PrimalStatus())
+        @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+
+        @test MOI.cangetattribute(m, MOI.ObjectiveValue())
+        @test MOI.getattribute(m, MOI.ObjectiveValue()) ≈ 2 atol=ε
+
+        @test MOI.cangetattribute(m, MOI.VariablePrimal(), v)
+        @test MOI.getattribute(m, MOI.VariablePrimal(), v) ≈ [0, 0, 1] atol=ε
+
+        @test MOI.cangetattribute(m, MOI.ConstraintPrimal(), c)
+        @test MOI.getattribute(m, MOI.ConstraintPrimal(), c) ≈ 1 atol=ε
+
+        if MOI.getattribute(solver, MOI.SupportsDuals())
+            @test MOI.cangetattribute(m, MOI.DualStatus())
+            @test MOI.getattribute(m, MOI.DualStatus()) == MOI.FeasiblePoint
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), c)
+            @test MOI.getattribute(m, MOI.ConstraintDual(), c) ≈ 2 atol=ε
+
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), vc1)
+            @test MOI.getattribute(m, MOI.ConstraintDual(), vc1) ≈ -1 atol=ε
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), vc2)
+            @test MOI.getattribute(m, MOI.ConstraintDual(), vc2) ≈ -2 atol=ε
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), vc3)
+            @test MOI.getattribute(m, MOI.ConstraintDual(), vc3) ≈ 0 atol=ε
+        end
+
+        # setting lb of x to -1 to get :
+        # max x + 2z
+        # s.t. x + y + z <= 1
+        # x >= -1
+        # y,z >= 0
+
+        MOI.modifyconstraint!(m, vc1, MOI.GreaterThan(-1.0))
+
+        MOI.optimize!(m)
+
+        @test MOI.cangetattribute(m, MOI.TerminationStatus())
+        @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+
+        @test MOI.cangetattribute(m, MOI.PrimalStatus())
+        @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+
+        @test MOI.cangetattribute(m, MOI.ObjectiveValue())
+        @test MOI.getattribute(m, MOI.ObjectiveValue()) ≈ 3 atol=ε
+
+        # put lb of x back to 0 and fix z to zero to get :
+        # max x + 2z
+        # s.t. x + y + z <= 1
+        # x, y >= 0, z = 0
+
+        MOI.modifyconstraint!(m, vc1, MOI.GreaterThan(0.0))
+        MOI.modifyconstraint!(m, vc3, MOI.EqualTo(0.0))
```

Comment:
> Do we really want to have this restriction?

If we allow changing the set, we have to allow changing the function as well. Then `ConstraintReference` can't have any type parameters, which will force solver interfaces to manage their own mappings of constraint references to classes of constraints. Having the constraint references properly typed will make it much easier to implement solver wrappers.

> In case we keep the restriction Why not delete this constraint and add and EqualTo constraint instead?

That's also ok.